### ISSUE TITLE
Support mj-wrapper nesting

### ISSUE THREAD

**Issue by caseyjhol** [mjmlio/mjml#2461]
**Is your feature request related to a problem? Please describe.**
I'd like the body of my emails to have a background color. In Outlook for Windows and MacOS, setting `background-color` on the body of the email results in the background color being applied when replying to or forwarding the email. This is undesired behavior.

A workaround is to wrap the email contents in a full width table and setting the background color there instead (https://stackoverflow.com/questions/23410279/how-to-remove-background-color-from-reply-of-html-email-template-in-outlook-2010). This doesn't work with MJML since we can't include MJML inside a table. I'm already using mj-wrapper in my email, so I can't set the background color there instead. If `mj-wrapper` could be nested, I could simply include everything inside a full-width `mj-wrapper` and use the background-color attribute.

This is what it looks like when replying to an email that has a background color set on the body:
![image](https://user-images.githubusercontent.com/2874325/163420792-6285ff37-dc47-4058-84d5-6d9f4347e807.png)

**Describe the solution you'd like**
Support nesting `mj-wrapper` inside `mj-wrapper`.

**Describe alternatives you've considered**
I tried setting the background color by styling `body > div` and setting the color there, but this results in no background color in Outlook for Windows.

**Additional context**
This might be kind of a fringe case, but I see no reason this shouldn't be supported. Ultimately it results in nesting a table inside a table, so I don't see how this could cause an issue.

Example of this working: https://mjml.io/try-it-live/7kMQYcEkP

**Comment by iRyusa** [comment]
Well as MJML is used mostly for Marketing/Transactionnal stuff I wasn't aware that was an issue.

`mj-wrapper` can be nested ( I think mj-section+column can be too ) but this is not properly tested so that's why the linter will say it's illegal.

The only issue I have in mind is with background image where full-width wrapper does some really weird stuff in Outlook.

So if it render fine in your target clients you can use it 👍  I don't think we have enough time for now to properly check every case that would imply to nesting wrapper

**Comment by caseyjhol** [comment]
@iRyusa Thanks for responding so quickly. I guess I'm mainly running into an issue where I'd like to be able to use nested `mj-wrapper` *and* use `mj-validator` without it throwing errors at me. There's currently no easy way to modify the dependencies for validator to tell it that nested `mj-wrapper` is allowed without a hacky workaround.

The fix here is just to add `'mj-wrapper'` as a supported child of `'mj-wrapper'` in `mjml-preset-core/src/dependencies.js`. I can open a PR if you'd be open to merging it?

**Comment by iRyusa** [comment]
It requires so much testing that I'm not really open into merging this. You
can just do as here
https://github.com/mjmlio/mjml/blob/bd9bb7f10235ab473a0279c2768161504eace326/packages/mjml/src/index.js#L6
and assign your own dep for mj-wrapper ?

**Comment by caseyjhol** [comment]
@iRyusa How do I use that? I'm not using gulp. Is it possible to set dependencies in mjmlconfig? My current workaround is to create an empty component file and register the component in mjmlconfig, but it's not ideal:

```js
const {BodyComponent} = require('mjml-core');
const {registerDependencies} = require('mjml-validator');

registerDependencies({
    'mj-wrapper': ['mj-wrapper'],
});

exports.default = BodyComponent;
```

**Comment by caseyjhol** [comment]
Alternatively, we may want to reconsider the way in which MJML sets body background color (we could simply wrap everything in a table/td). This is what it looks like when replying to an email in Outlook for MacOS (I've blurred some content):

![image](https://user-images.githubusercontent.com/2874325/164280268-ccb9cc85-a6f0-41ee-9ad5-369937c562df.png)

This email is supposed to have a solid gray background, but instead is all blocky and not at all what I would expect.

**Comment by iRyusa** [comment]
Yup for now I think you can only tricks it with a fake custom component. 

If we do this then it becomes a breaking change and need to ensure that everything is still rendering fine :/
